test/client/dio_interceptor_test.dart

```diff
@@ -0,0 +1,103 @@
+import 'dart:io';
+
+import 'package:dio/dio.dart';
+import 'package:http_mock_adapter/http_mock_adapter.dart';
+
+import 'package:requests_signature_dart/src/client/requests_signature_options.dart';
+import 'package:test/test.dart';
+import 'package:requests_signature_dart/requests_signature_dart.dart';
+
+void main() {
+  test('Interceptor adds signature header to GET request', () async {
+    // Arrange
+    final options = RequestsSignatureOptions(
+      clientId: 'test_client_id',
+      clientSecret: 'test_client_secret',
+      headerName: 'X-Signature',
+      signaturePattern: '{ClientId}:{Nonce}:{Timestamp}:{SignatureBody}',
+    );
+
+    final dio = Dio();
+
+    final interceptor = RequestsSignatureInterceptor(options, dio);
+
+    dio.interceptors.add(interceptor);
+
+    // Act
+    final response = await dio.get('https://google.ca');
+
+    // Assert
+    expect(response.statusCode, 200);
+
+    expect(response.requestOptions.headers, contains('X-Signature'));
+  });
+
+  test('Interceptor auto-retries on clockskew', () async {
```

Comment:
You need more than one test to validate the behavior on the clock skew.
I see at least 3 cases that need to be tested:
- You need to validate that the auto-retry doesn't work when the clock skew option is disabled.
- You need to validate that the auto-retry doesn't work when the clock skew option is enable but the tolerance is too low.
- You need to validate that the auto-retry works when the clock skew option is enable and the tolerance is high enough.